### ISSUE TITLE
Data Object Grid - Batch Editing

### ISSUE THREAD

**Issue by fashxp** [pimcore/studio-ui-bundle#302]
see also https://github.com/pimcore/studio-ui-bundle/issues/174

![image](https://github.com/pimcore/studio-ui-bundle/assets/8792145/0cfe5948-22ab-4a13-8638-dcceda6b7d96)

![image](https://github.com/pimcore/studio-ui-bundle/assets/8792145/15baa87d-84dd-4e0b-b3b7-7675dcbad436)

![image](https://github.com/pimcore/studio-ui-bundle/assets/8792145/a0046ee3-9eef-4369-b8bb-99ba2c155ac5)

https://www.figma.com/design/kTxowT29eCHwNh5NspDGGu/Pimcore-Components?node-id=2984-127260&t=090OJTL3MIaVURas-4

**Comment by lukmzig** [comment]
@markus-moser 
We discussed the batch editing of data objects and we would like to use Patch endpoint for it. Based on the design its not clear if this would be suitable for all field types. What do you think? 

We can have a further discussion if its not clear.

**Comment by markus-moser** [comment]
@lukmzig Yes it's fine when we use the patch endpoint for the batch editing (so the same approach like for assets). The batch edit modal will use the same frontend components like the editor and therefore the format will be the same.

**Comment by lukmzig** [comment]
As discussed for BE:
- Add endpoint for Data Objects similiar to `/pimcore-studio/api/assets/folder` for patch all objects
- Add new adapter interface for patching (Multi relation fields, multiple choice options, opt. Gallery) which accept "mode" - add or remove values (replace is default)
- As fallback use `getDataForSetter`  method

**Comment by lukmzig** [comment]
As discussed, some data object fields require additional data for Patching (e.g action). This applies to:

- all relation fields
- multi selection 
- image gallery

possible actions are:
- `add` - add values to existing values (if item is present but some metadata changed, then these will be updated)
- `remove` - remove some values from existing values
- `replace` / `null` - if null action or replace action is provided, values are replaced completely by new value (similiar to update behavior)

when patching these fields, please adhere to relevant data structure, e.g:

![Image](https://github.com/user-attachments/assets/0307e02c-e5f5-41b6-9d85-44ad9a22acc1)

or

![Image](https://github.com/user-attachments/assets/1a0f39b5-a2b1-42d3-8516-900a9b7b501c)

**Comment by markus-moser** [comment]
In general this is done. Let's create follow-up issues for topics which we find during testing.
